### DISCUSSION TITLE
Multiple checkbox in form

### DISCUSSION THREAD

**Discussion by ghost** [symfony/symfony#43935] [Q&A] [answered]
Hello, I want to add in my form a list of checkboxes that contain for each one the label of a subject (in my database). Here is my code
`public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        $builder
            ->add('email', EmailType::class)
            ->add('password', PasswordType::class)
            ->add('nom', TextType::class)
            ->add('prenom', TextType::class)
            ->add('specialite', TextType::class, ['required' => false])
            ->add('matieres', CheckboxType::class, ['required' => false])
            ->add('responsableFormateur', CheckboxType::class, ['required' => false])
            ->add('enregistrer', SubmitType::class)
        ;
    }`

`class Intervenant extends User
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
//    private $id;

    /**
     * @ORM\Column(type="string", length=50)
     */
    private $nom;

    /**
     * @ORM\Column(type="string", length=50)
     */
    private $prenom;

    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     */
    private $specialite;

    /**
     * @ORM\Column(type="boolean", nullable=true)
     */
    private $responsableFormateur;

    /**
     * @ORM\ManyToMany(targetEntity=Matiere::class, inversedBy="intervenants")
     */
    private $matieres;`

`class Matiere
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=50)
     */
    private $libelle;

    /**
     * @ORM\Column(type="integer", length=2)
     */
    private $heure;

    /**
     * @ORM\ManyToMany(targetEntity=Intervenant::class, mappedBy="matieres")
     */
    private $intervenants;`

`{{ form_start(intervenantForm, { 'attr': {'class': 'row g-3'} }) }}
        <div class="col-md-6">
            {{ form_row(intervenantForm.nom, { 'attr': {'class': 'form-control'} }) }}
        </div>
        <div class="col-md-6">
            {{ form_row(intervenantForm.prenom, { 'attr': {'class': 'form-control'} }) }}
        </div>
        <div class="col-12">
            {{ form_row(intervenantForm.email, { 'attr': {'class': 'form-control'} }) }}
        </div>
        <div class="col-12">
            {{ form_row(intervenantForm.password, { 'attr': {'class': 'form-control'}, 'value': password }) }}
        </div>
        <div class="col-md-3">
            {{ form_row(intervenantForm.specialite, { 'attr': {'class': 'form-control'} }) }}
        </div>
        <div class="col-md-6">
            Here for the matiere
        </div>
        <div class="col-md-3">
            {{ form_row(intervenantForm.responsableFormateur, { 'attr': {'class': 'form-check-input'} }) }}
        </div>
        <div class="col-12">
            {{ form_row(intervenantForm.enregistrer, { 'attr': {'class': 'btn btn-primary'} }) }}
        </div>
        {{ form_end(intervenantForm) }}`

Thank you

**Comment by ker0x** [comment, marked as answer]
You have to use the [EntityType](https://symfony.com/doc/current/reference/forms/types/entity.html) with options `expanded` and `multiple` set to `true` instead of the `CheckboxType`
